internal/app/frontend/frontend_service.go

```diff
@@ -45,32 +57,98 @@ func newFrontend(cfg config.View) (*frontendService, error) {
 	return fs, nil
 }
 
-// CreateTicket will create a new ticket, assign a Ticket id to it and put the
-// Ticket in state storage. It will then look through the 'properties' field
-// for the attributes defined as indices the matchmakaking config. If the
-// attributes exist and are valid integers, they will be indexed. Creating a
-// ticket adds the Ticket to the pool of Tickets considered for matchmaking.
+// CreateTicket will create a new ticket, assign an id to it and put it in state
+// storage. It will index the Ticket attributes based on the indexing configuration.
+// Indexing a Ticket adds the it to the pool of Tickets considered for matchmaking.
 func (s *frontendService) CreateTicket(ctx context.Context, req *pb.CreateTicketRequest) (*pb.CreateTicketResponse, error) {
-	return &pb.CreateTicketResponse{}, nil
+	// Perform input validation.
+	if req.Ticket == nil {
```

Comment:
The `req` itself could be nil, but grpc protocol does not allow any clients to send nil request so I think this check is not needed.

Reply:
Yes - but this check is not checking if request is nil - it is checking if ticket inside of the request is nil. Wont &pb.CreateTicketRequest{} be a valid request (not nil - but empty) - and still have ticket as a nil in there?

Reply:
Ah you are right, but it seems like we are missing this nil check everywhere.. maybe worth to send a separate pr to fix them.